include/RAJA/pattern/teams/teams_openmp.hpp

```diff
@@ -104,6 +104,76 @@ struct LoopExecute<omp_parallel_for_exec, SEGMENT> {
   }
 };
 
+  //
+  // Return local index
+  //
+template <typename SEGMENT>
+struct LoopICountExecute<omp_parallel_for_exec, SEGMENT> {
+
+  template <typename BODY>
+  static RAJA_INLINE RAJA_HOST_DEVICE void exec(
+      LaunchContext const RAJA_UNUSED_ARG(&ctx),
+      SEGMENT const &segment,
+      BODY const &body)
+  {
+
+    int len = segment.end() - segment.begin();
+#pragma omp parallel for
+    for (int i = 0; i < len; i++) {
+
+      body(*(segment.begin() + i), i);
+    }
+  }
+
+  template <typename BODY>
+  static RAJA_INLINE RAJA_HOST_DEVICE void exec(
+      LaunchContext const RAJA_UNUSED_ARG(&ctx),
+      SEGMENT const &segment0,
+      SEGMENT const &segment1,
+      BODY const &body)
+  {
+
+    const int len1 = segment1.end() - segment1.begin();
+    const int len0 = segment0.end() - segment0.begin();
+
+#pragma omp parallel for RAJA_COLLAPSE(2)
```

Comment:
Do we always want collapse here? Collapsing can improve or degrade performance depending on the loop lengths.

Reply:
That is a fair point, one option could be to add collapse policies, but in general I'm not sure what the API for perfectly nested loops in teams should look like. At the moment users can express perfectly nested loops by nesting RAJA::loops like so: 

```
RAJA::loop<pol1>(Range1){
 RAJA::loop<pol2>(Range2){
  RAJA::loop<pol3>(Range3){
 }
 }
}
```

Or we can support an interface that takes in a policy (or policies) ? With multiple segments? We have some support for this at the moment, but super experimental at the moment.  

```
RAJA::loop<pol3>(Range1, Range2, Range3) {} 
```

Reply:
@ajkunen or @trws do you guys have thoughts on how best to handle loop collapse as a user-specified choice? I don't think it is wise to always collapse the loops.

Reply:
We're currently doing it with multiple segments and a policy in kernel, so if RAJA users are used to anything it would probably be that.  If we want to keep the single segment setup though, which does have non-trivial advantages, we could go more the TBB or SYCL style route and make a couple of multi-dimensional segments.  Something like:

```c++
RAJA::loop<pol>(RAJA::range2d(len1, len2), [](RAJA::range2idx idx){
// idx.x, idx.y
});
// or
RAJA::loop<pol>(RAJA::range2d(len1, len2), RAJA::unpack([](int I, int j){});
// or 
RAJA::loop<pol>(RAJA::range2d(len1, len2), RAJA::unpack([](int I, int j){}); // slightly harder
```

Reply:
I like the idea of multi-dimensional segments, it would fit well with the application I work on.

Reply:
That would be relatively easy to implement too, at least for a few bounded dimension counts.  For others we could make an N-dimensional version that hands out a tuple or similar, but it would be syntactically nastier.  Will have to give a bit of thought to some of the implementation details to get all the behavior you probably want, but may make life easier for a few patterns.